addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestBridgeHandler.java

```diff
@@ -0,0 +1,332 @@
+        this.httpClient = new HttpClient(sslContextFactory);
+        this.httpClient.setConnectTimeout(30000);
+
+        NestBridgeConfiguration config = getConfigAs(NestBridgeConfiguration.class);
+        startAutomaticRefresh(config.refreshInterval);
+        this.accessToken = new NestAccessToken(config, this.httpClient);
+
+        logger.debug("Client Id       {}.", config.clientId);
+        logger.debug("Client Secret   {}.", config.clientSecret);
+        logger.debug("Pincode         {}.", config.pincode);
+        try {
+            logger.debug("Access Token    {}.", accessToken.getAccessToken());
+        } catch (InterruptedException e) {
+            logger.debug("Error getting Access Token.", e);
+        } catch (TimeoutException e) {
+            logger.debug("Error getting Access Token.", e);
+        } catch (ExecutionException e) {
+            logger.debug("Error getting Access Token.", e);
+        }
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Starting poll query");
+    }
+
+    /**
+     * Do something useful when the configuration update happens. Triggers changing
+     * polling intervals as well as re-doing the access token.
+     */
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        logger.info("Config update");
+        super.handleConfigurationUpdate(configurationParameters);
+        accessToken = new NestAccessToken(getConfigAs(NestBridgeConfiguration.class), this.httpClient);
+
+        try {
+            logger.debug("New Access Token {}.", accessToken.getAccessToken());
+        } catch (InterruptedException e) {
+            logger.debug("Error getting Access Token.", e);
+        } catch (TimeoutException e) {
+            logger.debug("Error getting Access Token.", e);
+        } catch (ExecutionException e) {
+            logger.debug("Error getting Access Token.", e);
+        }
+        stopAutomaticRefresh();
+        startAutomaticRefresh(getConfigAs(NestBridgeConfiguration.class).refreshInterval);
+    }
+
+    /**
+     * Clean up the handler.
+     */
+    @Override
+    public void dispose() {
+        logger.debug("Nest bridge disposed");
+        stopAutomaticRefresh();
+        this.accessToken = null;
+        this.lastDataQuery = null;
+        this.pollingJob = null;
+        this.pollingRunnable = null;
+        assert (this.listeners.isEmpty());
+    }
+
+    /**
+     * Handles an incoming command update
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // TODO Auto-generated method stub
+        if (command instanceof RefreshType) {
+            logger.debug("Refresh command received");
+            refreshData();
+        }
+    }
+
+    /**
+     * Read the data from nest and then parse it into something useful.
+     */
+    private void refreshData() {
+        String uri = "unknown";
```

Comment:
'unknown' is not a proper uri, it does not feel like a sane initialization. If `buildQueryString(config);` does not throw any exception maybe just move `String uri = buildQueryString(config);` out of try-catch